doc/src/base/collections.md

```diff
@@ -235,6 +236,60 @@ Partially implemented by:
   * [`Iterators.Pairs`](@ref)
 
 ## Set-Like Collections
+A [`Set`](@ref) is a collection of elements with no duplicated elements and keys/indices (unlike `Array` or `Dict`). So in a `Set` you can have only one of each element, and the order of elements isn't important.
+
+```jldoctest set_collections
+julia> colors = Set(["yellow", "blue", "green", "red"])
+Set{String} with 4 elements:
+  "yellow"
+  "blue"
+  "green"
+  "red"
+```
+
+Use [`push!`](@ref) to add elements to a set:
+```jldoctest set_collections
+julia> push!(colors, "black")
+Set{String} with 5 elements:
+  "yellow"
+  "blue"
+  "green"
+  "black"
+  "red"
+```
+
+Since sets don't store repeated elements, `push!` will not have any effect for pushing the elements that are already in the set. Also because `Set` doesn't have a concept of "first" (unlike `Array`), using [`pushfirst!`](@ref) to insert an item at the beginning of a set will result in an error.
+
+To see if something is in the set, use [`in`](@ref):
+```jldoctest set_collections
+julia> "green" in colors  # or: in("green", colors)
+true
+```
+
+To access each element of a `Set`, you can loop over the elements:
+```jldoctest set_collections
+julia> for color in colors
+  println(color)
+end
```

Comment:
This doctest is badly indented. It should be:

```
julia> for color in colors
         println(color)
       end

Reply:
Thank you! I was confused by this.